proto/p4/config/v1/p4types.proto

```diff
@@ -183,4 +185,22 @@ message P4ErrorTypeSpec {
   repeated string members = 1;
 }
 
+message P4NewTypeTranslation {
+  // the URI uniquely identifies the translation in order to enable the
+  // P4Runtime agent to perform value-mapping appropriately when required. It is
+  // recommended that the URI includes at least the P4 architecture name and the
+  // type name.
+  string uri = 1;
+  int32 sdn_bitwidth = 2;
+}
+
+// New types introduced with the "type" keyword
+message P4NewTypeSpec {
+  P4DataTypeSpec original_type = 1;
```

Comment:
One potential issue is that `original_type` may be different across different targets / backends (e.g. if the device port representation is different). To keep P4Info target independent, it may be better to omit `original_type` for translated entities.

Reply:
I mentioned this in another comment, but also replying here to keep it with the previous comment: perhaps define a unique and easily recognized value for P4DataTypeSpec that means "target-dependent"?  So far as I know right now that value would only be legal in the context of the value of the `original_type` key for one of the PSA target-dependent type names, and would be nonsense to use that value in most or all other contexts.

Reply:
I thought it might make more sense to use a union (Protobuf `oneof`) so that we either include `original_type` (in the absence of translation) or `translated_type` (in the case of translation). I have made the change.